src/main/kotlin/com/example/delivery/config/LogInvokeAnnotationBeanPostProccessor.kt

```diff
@@ -0,0 +1,95 @@
+package com.example.delivery.config
+
+import com.example.delivery.annotaion.InvokeLog
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.cglib.proxy.Enhancer
+import org.springframework.cglib.proxy.MethodInterceptor
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Component
+import java.lang.reflect.Method
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.jvm.isAccessible
+import kotlin.reflect.jvm.jvmErasure
+
+@Component
+@Order(Ordered.HIGHEST_PRECEDENCE)
+class LogInvokeAnnotationBeanPostProcessor : BeanPostProcessor {
+
+    private val beans = HashMap<String, AnnotationTarget>()
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any? {
+        val beanClass = bean.javaClass
+        when {
+            beanClass.isAnnotationPresent(InvokeLog::class.java) -> {
+                beans[beanName] = AnnotationTarget.CLASS
+            }
+
+            beanClass.declaredMethods.any { it.isAnnotationPresent(InvokeLog::class.java) } -> {
+                beans[beanName] = AnnotationTarget.FUNCTION
+            }
+        }
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
+        if (beans.containsKey(beanName)) {
+            return createProxy(bean, beanName)
+        }
+        return bean
```

Comment:
Minimize return statements
```suggestion
        return if (beans.containsKey(beanName)) {
            createProxy(bean, beanName)
        } else {
            bean
        }        
```